src/watch.rs

```diff
@@ -41,6 +48,8 @@ pub fn watch(
                 }
             );
 
+            suspend.swap(true, Ordering::SeqCst);
```

Comment:
You are not using the previous value, so `suspend.store(true, Ordering::SeqCst);` should suffice here and below.